graphbolt/src/columnwise_pick.h

```diff
@@ -0,0 +1,139 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file columnwise_pick.h
+ * @brief Contains the methods definition for column wise pick.
+ */
+
+#ifndef GRAPHBOLT_COLUMNWISE_PICK_H_
+#define GRAPHBOLT_COLUMNWISE_PICK_H_
+
+#include <graphbolt/csc_sampling_graph.h>
+#include <graphbolt/sampled_subgraph.h>
+
+namespace graphbolt {
+namespace sampling {
+
+/**
+ * @brief Alias for a function type that takes start, end, and num_samples as
+ * arguments and returns a torch Tensor.
+ *
+ * @param start The starting index for picking elements.
+ * @param end The ending index for picking elements.
+ * @param num_samples The number of samples to be picked.
+ * @return A torch Tensor containing the picked elements.
+ */
+using RangePickFn = std::function<torch::Tensor(
+    int64_t start, int64_t end, int64_t num_samples)>;
+
+/**
+ *  @brief  Alias for a vector of torch Tensors.
+ */
+using TensorList = std::vector<torch::Tensor>;
+
+/**
+ * @brief Default pick grain size for parallelized picking.
+ */
+static constexpr int kDefaultPickGrainSize = 32;
+
+/**
+ * @brief Picks random values uniformly from a range [start, end) with
+ * replacement.
+ *
+ * @param start The start of the range (inclusive).
+ * @param end The end of the range (exclusive).
+ * @param num_samples The number of samples to pick.
+ * @return A tensor containing the picked values.
+ */
+inline torch::Tensor UniformPickWithReplace(
```

Comment:
It is strange to see an `inline` function declaration in a header because if some source file include it without explicit definition, the linker will fail. In general, either inline function is directly defined in the header (as the name suggested) or is placed in the source file.